View/View.php

```diff
@@ -132,6 +133,21 @@ public function setData($data)
     }
 
     /**
+     * Set template variable
+     *
+     * @param $variableName
+     * @param $data
+     *
+     * @return View
+     */
+    public function setTemplateData($variableName, $data)
```

Comment:
I know it is not required, but IMO is more consistent with current interface (only one 'array' passed to `view`/`setData`) and will guide the developer to a cleaner code. Also, if you allow setting one variable at a time as in `Request` object, you need to implement a container interface or it will be impossible to unset properties.

Your example can be refactored as:
```php
$viewData = array('category' => $category);
/* more code */
$viewData['other_template_data'] = $otherTemplateData;

return $this->view($products, 200)
    ->setTemplate("SiteViewBundle:Category:show.html.twig")
    ->setTemplateVar('products')
    ->setTemplateData($viewData)
;
```
or even better with the lazy approach:
```php

return $this->view($products, 200)
    ->setTemplate("SiteViewBundle:Category:show.html.twig")
    ->setTemplateVar('products')
    ->setTemplateData(function () use ($category, $container) {
        $viewData = array('category' => $category);
        /* more code using services & co */
        $viewData['other_template_data'] = $otherTemplateData;
        // adding 'products' key not recommended :)
        return $viewData;
    })
;
```